pandas/core/groupby/generic.py

```diff
@@ -11,13 +11,15 @@
 from functools import partial
 from textwrap import dedent
 from typing import (
+    TYPE_CHECKING,
     Any,
     Callable,
     Hashable,
     Iterable,
     Mapping,
     NamedTuple,
     Sequence,
+    Tuple,
```

Comment:
we cant just use tuple?

Reply:
I tried to do that but was getting some error (unfortunately don't remember which one exactly). However, it seems that it will no longer be needed as it was only imported for line 629 but it will be left as it was in this PR

Reply:
Yes, `tuple` can only be used in type annotations (function and variable). `cast` is actually a runtime construct that is a no-op but does need to be parsed by the interpreter nonetheless, so must import `Tuple` from typing in these cases.

> However, it seems that it will no longer be needed as it was only imported for line 629 but it will be left as it was in this PR

yep, just mentioned the above for clarification